### PR TITLE
HS-598/601 passing access token to inventory, verify access token and get tenant id  in the inventory service

### PR DESCRIPTION
**Pull request by yangli2014**
## Description

Test with GraphQL playground at http://localhost:8123/api/gui

- All the gRPC-enabled query APIs will start **find**
- addNode is the new gRPC-enabled mutation to add a new node.

## Jira link(s)
- https://issues.opennms.org/browse/HS-598
- https://issues.opennms.org/browse/HS-601

## Checklist
* [x] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [x] Appropriate reviewer(s) have been selected.
* [x] Jira issue(s) have been updated to "In Review".
* [x] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.

### CODE REVIEW

File: inventory/src/main/java/org/opennms/horizon/inventory/component/MinionHeartbeatConsumer.java [hunk 1]
```diff
@@ -46,11 +52,12 @@
 public class MinionHeartbeatConsumer {
     private final MonitoringSystemService service;
     @KafkaListener(topics = "${kafka.topics.minion-heartbeat}", concurrency = "1")
-    public void receiveMessage(byte[] data) {
+    public void receiveMessage(@Payload byte[] data, @Headers Map<String, Object> headers) {
         try {
             HeartbeatMessage message = HeartbeatMessage.parseFrom(data);
             log.info("Received heartbeat message for minion with id {} and location {}", message.getIdentity().getSystemId(), message.getIdentity().getLocation());
-            service.addMonitoringSystemFromHeartbeat(message);
+            String tenantId = Optional.ofNullable(headers.get(Constants.TENANT_ID_KEY)).map(o -> (String)o).orElse(Constants.DEFAULT_TENANT_ID);
```
Comment (cgorantla) [inline inventory/src/main/java/org/opennms/horizon/inventory/component/MinionHeartbeatConsumer.java:59]: i think the header value Object is always a byte[], see [this](https://github.com/OpenNMS/horizon-stream/pull/465/files#diff-c677224db581958011f2476fe00f9caaa7a20f52c11897df85e370579131acc3R118)

File: inventory/src/main/java/org/opennms/horizon/inventory/grpc/InventoryServerInterceptor.java [hunk 2]
```diff
@@ -28,30 +28,63 @@
...
+        }
+        catch (NoSuchElementException e) {
             serverCall.close(Status.UNAUTHENTICATED.withDescription("Missing tenant id"), new Metadata());
             return new ServerCall.Listener<>() {};
         }
-        Context context = Context.current().withValue(TENANT_ID, tenantId);
-        return Contexts.interceptCall(context, serverCall, headers, callHandler);
     }
+
+    protected Optional<String> verifyAccessToken(String authHeader) throws VerificationException {
```
Comment (cgorantla) [inline inventory/src/main/java/org/opennms/horizon/inventory/grpc/InventoryServerInterceptor.java:77]: i may be missing the context here but i think the idea was that this verification and parsing of tenant id happens on BFF and  Inventory service will get `tenant-id` directly in the headers.

**Review by cgorantla — COMMENTED** [review]
Nice work!

I think inventory service doesn't need to deal with parsing of access token, I think this should be done in BFF.  Please confirm with @geraldhumphries !

File: inventory/src/main/java/org/opennms/horizon/inventory/grpc/InventoryServerInterceptor.java (hunk 2 above)
Comment (yangli2014) [reply inventory/src/main/java/org/opennms/horizon/inventory/grpc/InventoryServerInterceptor.java:77]: As we discussed the other day, the access token will be passed to downstream services, and each service will verify and validate the token and then get the tenant id.

**Pull request merged**